MATLAB/rl_siso_NEW.m

```diff
@@ -0,0 +1,496 @@
+    rx_vec_iris_tmp = H_ul.*tx_vec_iris.' + W_ul;
+    rx_vec_iris_tmp = rx_vec_iris_tmp.';
+
+    numGoodFrames = 1;  % Don't care about number of frames in SIM MODE
+else
+    % Set up the Iris experiment
+    disp("Running: HARDWARE MODE");
+
+    % Create two Iris node objects:
+    tx_direction = 'uplink';      % Options: {'uplink', 'downlink', 'hub_sync'}
+    bs_ids = ["RF3E000722"];
+    ue_ids = ["RF3E000665"];
+    hub_id = ["FH4B000003"];
+
+    % Iris nodes' parameters
+    sdr_params = struct(...
+        'bs_id', bs_ids, ...
+        'ue_id', ue_ids,...
+        'hub_id', hub_id,...
+        'bs_ant', ANT_BS, ...
+        'ue_ant', ANT_UE, ...
+        'txfreq', TX_FRQ, ...
+        'rxfreq', RX_FRQ, ...
+        'txgain', TX_GN, ...
+        'tx_gain_ue', TX_GN_UE, ...
+        'rxgain', RX_GN, ...
+        'sample_rate', SMPL_RT);
+
+    mimo_handle = mimo_driver(sdr_params);
+
+    if FIND_OPTIMAL_GAINS
+        [txg_opt, rxg_opt, max_num_beacon, valid] = mimo_handle.mimo_set_opt_gains(N_FRM);
+        fprintf(" !!! Optimal Gain Setting: TX %d, RX %d, max_num_beacon %d !!! \n", txg_opt, rxg_opt, max_num_beacon);
+        if ~valid
+            printf("[ERROR] Could NOT find an adequate gain setting (no beacons detected). Check setup!");
+            return;
+        end
+        mimo_handle.mimo_update_sdr_param('txgain', txg_opt);
+        mimo_handle.mimo_update_sdr_param('rxgain', rxg_opt);
+    end
+
+    if strcmp(tx_direction, 'uplink')
+        [rx_vec_iris_tmp, numGoodFrames, ~] = mimo_handle.mimo_txrx_uplink(tx_vec_iris, N_FRM, N_ZPAD_PRE);
+
+    elseif strcmp(tx_direction, 'downlink')
+        [rx_vec_iris_tmp, numGoodFrames, ~] = mimo_handle.mimo_txrx_downlink(tx_vec_iris, N_FRM, N_ZPAD_PRE);
+
+    elseif strcmp(tx_direction, 'hub_sync')
+        if isempty(hub_id)
+            error('Hub ID must be specified in hub_sync transmission mode. Exit Now!');
+        end
+        bs_sched = ["GGGGGRG"];
+        ue_sched = ["GGGGGPG"];
+        [rx_vec_iris_tmp, numGoodFrames, ~] = mimo_handle.mimo_txrx_refnode(tx_vec_iris, N_FRM, bs_sched, ue_sched, N_ZPAD_PRE);
+    else
+        error('TX Method Not Supported. Exit Now!');
+    end
+
+    mimo_handle.mimo_close();
+
+end
+
+if (isempty(rx_vec_iris_tmp))
+    return;
+end
+
+% Process data
+for frm_idx = 1:numGoodFrames
+    fprintf(' =============================== \n');
+    fprintf('Frame #%d Out of %d Triggered Frames \n', frm_idx, numGoodFrames);
+
+    if numGoodFrames == 1
+        rx_vec_iris = squeeze(rx_vec_iris_tmp);
+    else
+        rx_vec_iris = squeeze(rx_vec_iris_tmp(frm_idx, 1, 1, :));
+    end
+
+    if DEBUG
+        figure; plot(abs(rx_vec_iris));
+    end
+    %% Correlate for LTS
+    % Complex cross correlation of Rx waveform with time-domain LTS
+    a = 1;
+    unos = ones(size(preamble.'))';
+    v0 = filter(flipud(preamble'),a,rx_vec_iris);
+    v1 = filter(unos,a,abs(rx_vec_iris).^2);
+    m_filt = (abs(v0).^2)./v1; % normalized correlation
```

Comment:
I have had issues with this match filtering scheme (consistently wrong offset). I am not sure if you have seen it. But we may want to do it differently.

Reply:
yeah, not a big deal with SISO but I'm doing it differently in other scripts. Need to check if I changed them all though